driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -235,17 +286,18 @@ public Boolean isIdempotent() {
         return !hasNonIdempotentOps();
     }
 
-    /**
-     * {@inheritDoc}
-     * <p>
-     * Note: Calling this method may trigger the underlying {@link Cluster} initialization.
-     */
     @Override
     public String toString() {
-        if (forceNoValues) {
-            return getQueryString();
+        try {
+            if (forceNoValues) {
+                return getQueryString();
+            }
+            StringBuilder queryString = buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE);
+            return maybeAddSemicolon(queryString).toString();
+        } catch (CodecNotFoundException e) {
+            // Ugly but we have absolutely no context to get the registry from
+            return String.format("built query (could not generate with default codec registry: %s)", e.getMessage());
```

Comment:
I think we can safely propose something better here:
```java
StringBuilder queryString = buildQueryString(new ArrayList<Object>(), CodecRegistry.DEFAULT_INSTANCE);
return maybeAddSemicolon(queryString).toString();
```
When the parameter `variables` is non null, in `Utils.appendValue`line 94 the only case the codec registry could get accessed is if `Utils.isSerializable(value)` returns false – which can only happen for a `Number` or a collection thereof. So for an unknown object, a bind marker would be inserted instead. E.g.:
```java
System.out.println(insertInto("foo").value("a", new Foo()));
```
This would print:
```
INSERT INTO foo (a) VALUES (?);
```